docs/python_docs/python/tutorials/getting-started/crash-course/2-create-nn.md

```diff
@@ -0,0 +1,558 @@
+ },
+ {
+  "data": {
+   "text/plain": "array([[ 9.5244788e-05,  2.9054983e-03, -6.9242873e-04,  2.2817445e-03,\n        -8.9324836e-04, -6.6934002e-04, -2.7392569e-03,  3.3438532e-04,\n        -4.0528120e-04,  2.8272711e-03]])"
+  },
+  "execution_count": 102,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+In this example, you can use `print` to get the intermediate results between
+layers. This approach provides a more flexible way to define the forward
+function.
+
+You can use `[]` to index a particular layer. For example, the following
+accesses the first layer's weight and sixth layer's bias.
+
+```{.python .input  n=108}
+net.conv1.weight.data().shape, net.dense1.bias.data().shape
+
+```
+
+```{.json .output n=108}
+[
+ {
+  "data": {
+   "text/plain": "((6, 1, 3, 3), (120,))"
+  },
+  "execution_count": 108,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+## Using predefined (pretrained) architectures
+
+Till now, we have seen how to create your own neural network architectures. But
+what if we want to replicate or baseline your dataset using some of the common
+models in computer visions or natural language processing (nlp). Gluon includes
+common architectures that you can directly use. The Gluon Model Zoo provides a
+collection of off-the-shelf models e.g. RESNET, BERT etc. Some of the common
+architectures include:
+
+- [Gluon CV model zoo](https://gluon-cv.mxnet.io/model_zoo/index.html)
+
+- [Gluon NLP model zoo](https://gluon-nlp.mxnet.io/model_zoo/index.html)
+
+Lets look at an example
+
+```{.python .input  n=117}
+from mxnet.gluon import model_zoo
+
+net = model_zoo.vision.resnet50_v2(pretrained=True)
+net.hybridize()
+
+dummy_input = np.ones(shape=(1,3,224,224))
+output = net(dummy_input)
+
+output.shape
+        
+```
+
+```{.json .output n=117}
+[
+ {
+  "data": {
+   "text/plain": "(1, 1000)"
+  },
+  "execution_count": 117,
+  "metadata": {},
+  "output_type": "execute_result"
+ }
+]
+```
+
+## Deciding the paradim for your NN
+
+MXNet includes two types of programming paradigms:
+
+1. Imperative API - Gluon that is very user friendly and allows for quick
+prototyping, easy debugging and natural control flow for people familiar with
+python programming.
+2. Symbolic or Declarative API for creating static graphs with low level
+optimizations on operators and allows for support in multiple languages, cross-
+compatibility with other frameworks (via ONNX) and as a backend for other high
+level APIs like keras. However, it’s less flexible because any logic must be
+encoded into the graph as special operators like scan, while_loop and cond. It’s
+also hard to debug.
```

Comment:
Let's not refer to Imperative / Symbolic "API" here, as in MXNet 2 there is only the numpy compatible API. In MXNet 2 we can use the numpy compatible API in an imperative, eager fashion or we can use it in the lazy, deferred compute mode which traces / records all calls to the numpy operators. Based on the recording / trace a symbolic graph is constructed in the backend. The activation of lazy, deferred compute mode for recording / trace is automatically handled by HybridBlock.hybridize()